src/java/com/wolfssl/provider/jsse/WolfSSLSocket.java

```diff
@@ -1901,9 +1906,13 @@ public void write(byte[] b, int off, int len) throws IOException {
                     }
                 }
 
-                /* do handshake if not completed yet, handles synchronization */
-                if (socket.handshakeComplete == false) {
-                    socket.startHandshake();
+                try {
+                    /* do handshake if not completed yet, handles synchronization */
+                    if (socket.handshakeComplete == false) {
+                        socket.startHandshake();
+                    }
+                } catch (SocketTimeoutException e) {
+                    throw new SocketTimeoutException("ssl.write in startHandshake");
```

Comment:
Same as above, should we add a debug log for the message, then `throw e` for the original exception?